homeassistant/components/tplink/button.py

```diff
@@ -25,9 +31,19 @@ class TPLinkButtonEntityDescription(
 BUTTON_DESCRIPTIONS: Final = [
     TPLinkButtonEntityDescription(
         key="test_alarm",
+        deprecated_info=DeprecatedInfo(
```

Comment:
I saw that you made the PR https://github.com/home-assistant/core/pull/125594 but for me it just seems overly complicated?
Should/Could `async_check_create_deprecated` not be a simple helper function which you can run the entities through to deprecate them instead of moving it into the entity description and thus making it more complex (which I don't think is needed)

Reply:
I created https://github.com/home-assistant/architecture/discussions/1133 because I think this is a common pattern that could/should be provided by core.

Reply:
I know. My question is still why is this not just a simple helper instead of building this complex thing?
As that discussion is not close to be finalized I think we still need to address it here as this PR is ongoing.

Reply:
Moved reply [to comment below](https://github.com/home-assistant/core/pull/126349#issuecomment-2367635291)

Reply:
I don't have strong feelings on where the `DeprecatedInfo` gets stored.  That should be worked out to a solution that everyone is ok with in https://github.com/home-assistant/architecture/discussions/1133 

For the `tplink` integration, its fine to move forward with this design and adapt to the final outcome of the arch discussion later.

Reply:
I'm fine with leaving that for the arch discussion to not block this PR 👍